Mathlib/Data/Multiset/Basic.lean

```diff
@@ -996,6 +996,11 @@ theorem card_erase_eq_ite {a : α} {s : Multiset α} :
   · rwa [card_erase_of_mem h, if_pos]
   · rwa [erase_of_not_mem h, if_neg]
 
+theorem in_of_ne_of_cons_erase {a a0: α} {M X : Multiset α}
+    (H : M = (a0 ::ₘ X).erase a) (hyp : ¬ a = a0) : a0 ∈ M := by
+  rw [H, Multiset.mem_erase_of_ne fun h ↦ hyp <| id <| Eq.symm h, Multiset.mem_cons]
+  tauto
```

Comment:
I don't think this lemma is worth having:
1. We almost never have lemmas taking an equality as an assumption. The reasons are that
  a. The user is (almost) always able to substitute that equality before trying to apply any lemma
  b. Adding all possible such lemmas would lead to combinatorial explosion
2. Once you substitute `M` in, the proof is structurally made of existing lemmas, namely `(mem_erase_of_ne hyp.symm).2 (mem_cons_self ..)`